libs/verifiable-credentials/src/main/java/uk/gov/di/ipv/core/library/verifiablecredential/helpers/VcHelper.java

```diff
@@ -51,54 +40,37 @@ public class VcHelper {
     private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
     private static final List<String> DL_UK_ISSUER_LIST = Arrays.asList("DVLA", "DVA");
     private static final String UK_ICAO_ISSUER_CODE = "GBR";
-    private static final CollectionType CREDENTIAL_EVIDENCE_ITEM_LIST_TYPE =
-            OBJECT_MAPPER
-                    .getTypeFactory()
-                    .constructCollectionType(List.class, CredentialEvidenceItem.class);
     private static ConfigService configService;
     private static final int ONLY = 0;
-    private static final List<String> OPERATIONAL_CRIS = Cri.getOperationalCriIds();
-    private static final List<String> NON_EVIDENCE_CRI_TYPES = Cri.getNonEvidenceCriIds();
 
     private VcHelper() {}
 
     public static void setConfigService(ConfigService configService) {
         VcHelper.configService = configService;
     }
 
-    public static boolean isSuccessfulVc(VerifiableCredential vc) throws CredentialParseException {
-        var evidenceArray =
-                OBJECT_MAPPER.valueToTree(vc.getClaimsSet().getClaim(VC_CLAIM)).path(VC_EVIDENCE);
-
-        var excludedCredentialIssuers = getNonEvidenceCredentialIssuers();
-
-        if (evidenceArray.isMissingNode() || evidenceArray.isEmpty()) {
-            String vcIssuer = vc.getClaimsSet().getIssuer();
-            if (excludedCredentialIssuers.contains(vcIssuer)) {
-                return true;
+    public static boolean isSuccessfulVc(VerifiableCredential vc) {
+        if (vc.getCredential() instanceof IdentityCheckCredential identityCheckCredential) {
+            var evidence = identityCheckCredential.getEvidence();
+            if (isNullOrEmpty(evidence)) {
+                LOGGER.warn(
+                        "Unexpected missing evidence on VC from issuer: {}",
+                        vc.getClaimsSet().getIssuer());
+                return false;
             }
-            LOGGER.warn("Unexpected missing evidence on VC from issuer: {}", vcIssuer);
-            return false;
-        }
-
-        try {
-            return isValidEvidence(
-                    OBJECT_MAPPER.treeToValue(evidenceArray, CREDENTIAL_EVIDENCE_ITEM_LIST_TYPE));
-        } catch (JsonProcessingException e) {
-            throw new CredentialParseException("Unable to create credential evidence list", e);
+            return evidence.stream()
+                    .anyMatch(
+                            check -> Gpg45IdentityCheckValidator.isSuccessful(check, vc.getCri()));
         }
+        return true;
```

Comment:
So anything that isn't an IdentityCheckCredential will be true? Such as an AddressCredential - which we won't run the validator on. But if we did run the validator on an address VC it'd be false?

Reply:
To be a bit more precise, an AddressCredential will always be 'successful', but an IdentityCheckCredential from the Address CRI will be 'unsuccessful'.

As above, perhaps this should be an error instead...

Reply:
I think that't the clarification I was missing. The warning log makes sense.